From df121e30406a1f892f383f86c76db936992354c2 Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Tue, 13 Aug 2013 00:02:18 +0200
Subject: [PATCH] isapc: disable kvmvapic

vapic requires the VAPIC ROM to be mapped into RAM.  This is not
possible without PAM hardware.  This fixes a segmentation fault
running with -M isapc.

Cc: qemu-stable@nongnu.org
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>

(crobinso: s/kvmvapic/vapic/g)
---
 hw/pc_piix.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/hw/pc_piix.c b/hw/pc_piix.c
index a1a6794..9f3f93b 100644
--- a/hw/pc_piix.c
+++ b/hw/pc_piix.c
@@ -689,7 +689,11 @@ static QEMUMachine isapc_machine = {
             .property = "rom_only",
             .value    = stringify(1),
         },
-        { /* end of list */ }
+        {
+            .driver   = "apic-common",
+            .property = "vapic",
+            .value    = "off",
+        },
     },
     DEFAULT_MACHINE_OPTIONS,
 };
